chore(ts): migrate tsconfig off deprecated baseUrl (TS5101)

## Problem

Running tsc against the root tsconfig emits:

```text
tsconfig.json(26,5): error TS5101: Option 'baseUrl' is deprecated and will stop functioning in TypeScript 7.0. Specify compilerOption '"ignoreDeprecations": "6.0"' to silence this error.
```

## Acceptance criteria

- Path aliases work without baseUrl (relative paths entries or SvelteKit alias config), or ignoreDeprecations is set as an explicit documented stopgap
- npx tsc --noEmit reports no TS5101
- Vite, Vitest, svelte-check, and editor resolution still resolve $lib and other aliases

## Notes

Surfaced 2026-06-11 during PR #2070 verification. Fits the M04 type-safety theme; should land before any TypeScript 7 upgrade.

--- a/src/lib/schemas/index.ts
+++ b/src/lib/schemas/index.ts
@@ -929,12 +929,15 @@ function recoverSlotPositions<
   const recovery = new Map<string, "left" | "right">();
   const recoveredSlugs = new Set<string>();
   for (const group of groups.values()) {
+    const [d0, d1] = group;
     if (
       group.length === 2 &&
+      d0 &&
+      d1 &&
       group.every((d) => d.slot_position === undefined)
     ) {
-      recovery.set(group[0].id, "left");
-      recovery.set(group[1].id, "right");
+      recovery.set(d0.id, "left");
+      recovery.set(d1.id, "right");
       for (const d of group) recoveredSlugs.add(d.device_type);
     }
   }

--- a/src/lib/stores/commands/device.ts
+++ b/src/lib/stores/commands/device.ts
@@ -446,8 +446,7 @@ export function createCrossRackMoveCommand(
 
       // 3. Place children in target rack with remapped container_id
       childPlacedIndices.length = 0;
-      for (let i = 0; i < placedChildren.length; i++) {
-        const child = placedChildren[i];
+      placedChildren.forEach((child, i) => {
         const childToPlace: PlacedDevice =
           child.container_id && child.container_id !== actualParentId
             ? { ...child, container_id: actualParentId }
@@ -458,11 +457,12 @@ export function createCrossRackMoveCommand(
         // Re-key child placement image if child ID was remapped (#1478)
         const actualChild = store.getDeviceAtIndex(idx);
         const actualChildId = actualChild?.id ?? childToPlace.id;
-        if (actualChildId !== currentChildImageIds[i]) {
-          rekeyPlacementImage(currentChildImageIds[i], actualChildId);
+        const previousChildImageId = currentChildImageIds[i];
+        if (previousChildImageId && actualChildId !== previousChildImageId) {
+          rekeyPlacementImage(previousChildImageId, actualChildId);
           currentChildImageIds[i] = actualChildId;
         }
-      }
+      });
 
       // 4. Restore active rack
       store.setActiveRackId(savedActiveRack);
@@ -495,8 +495,7 @@ export function createCrossRackMoveCommand(
       currentSourceDeviceIds[0] = undoActualParentId;
 
       // 3. Place children back in source rack with remapped container_id
-      for (let i = 0; i < childrenCopies.length; i++) {
-        const child = childrenCopies[i];
+      childrenCopies.forEach((child, i) => {
         const childToPlace: PlacedDevice =
           child.container_id && child.container_id !== undoActualParentId
             ? { ...child, container_id: undoActualParentId }
@@ -506,12 +505,13 @@ export function createCrossRackMoveCommand(
         // Re-key child placement image and update source ID tracking if remapped (#1478)
         const actualChild = store.getDeviceAtIndex(undoChildIdx);
         const actualChildId = actualChild?.id ?? childToPlace.id;
-        if (actualChildId !== currentChildImageIds[i]) {
-          rekeyPlacementImage(currentChildImageIds[i], actualChildId);
+        const previousChildImageId = currentChildImageIds[i];
+        if (previousChildImageId && actualChildId !== previousChildImageId) {
+          rekeyPlacementImage(previousChildImageId, actualChildId);
           currentChildImageIds[i] = actualChildId;
         }
         currentSourceDeviceIds[i + 1] = actualChildId;
-      }
+      });
 
       // 4. Restore active rack
       store.setActiveRackId(savedActiveRack);

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -23,6 +23,7 @@
     "forceConsistentCasingInFileNames": true,
 
     /* Path aliases */
+    "ignoreDeprecations": "6.0",
     "baseUrl": ".",
     "paths": {
       "$lib/*": ["src/lib/*"]

--- a/vitest.config.ts
+++ b/vitest.config.ts
@@ -44,7 +44,7 @@ function getMaxWorkers(): number {
 const maxWorkers = getMaxWorkers();
 
 export default defineConfig({
-  plugins: [svelte({ hot: !process.env.VITEST })],
+  plugins: [svelte()],
   define: {
     // Inject version at build time (same as vite.config.ts)
     __APP_VERSION__: JSON.stringify(pkg.version),

